diff --git a/software/rocketlogger/Makefile b/software/rocketlogger/Makefile
index 9729c3c..5299601 100644
--- a/software/rocketlogger/Makefile
+++ b/software/rocketlogger/Makefile
@@ -94,7 +94,6 @@ CXXSTD = -std=c++11
 
 ## libraries
 LIBS = -lprussdrv
-LIBS += -li2c
 LIBS += -lncurses
 LIBS += -lrt
 
diff --git a/software/rocketlogger/rl.c b/software/rocketlogger/rl.c
index 1467631..7b5cfb3 100644
--- a/software/rocketlogger/rl.c
+++ b/software/rocketlogger/rl.c
@@ -103,7 +103,19 @@ const char *RL_CHANNEL_VALID_NAMES[RL_CHANNEL_SWITCHED_COUNT] = {"I1L_valid",
                                                                  "I2L_valid"};
 
 /// Global RocketLogger status variable.
-rl_status_t rl_status = rl_status_default;
+rl_status_t rl_status = {
+    .sampling = false,
+    .error = false,
+    .sample_count = 0,
+    .buffer_count = 0,
+    .calibration_time = 0,
+    .calibration_file = RL_CALIBRATION_SYSTEM_FILE,
+    .disk_free = 0,
+    .disk_free_permille = 0,
+    .disk_use_rate = 0,
+    .sensor_count = 0,
+    .sensor_available = {false},
+}; // copy of rl_status_default, as GCC6 does not accept constant initializers
 
 /**
  * Print a configuration setting line with formated string value.
diff --git a/software/rocketlogger/sensor/bme280.c b/software/rocketlogger/sensor/bme280.c
index fe2becb..59d91be 100644
--- a/software/rocketlogger/sensor/bme280.c
+++ b/software/rocketlogger/sensor/bme280.c
@@ -31,7 +31,7 @@
 
 #include <stdint.h>
 
-#include <i2c/smbus.h>
+#include <linux/i2c-dev.h>
 
 #include "../log.h"
 #include "sensor.h"
diff --git a/software/rocketlogger/sensor/tsl4531.c b/software/rocketlogger/sensor/tsl4531.c
index 3df95da..82c405e 100644
--- a/software/rocketlogger/sensor/tsl4531.c
+++ b/software/rocketlogger/sensor/tsl4531.c
@@ -31,7 +31,7 @@
 
 #include <stdint.h>
 
-#include <i2c/smbus.h>
+#include <linux/i2c-dev.h>
 
 #include "../log.h"
 #include "sensor.h"
